packages/running-extension/src/kernels.tsx

```diff
@@ -224,20 +290,26 @@ namespace Private {
     }
 
     icon(): LabIcon | string {
-      const { spec } = this;
-      if (!spec || !spec.resources) {
-        return jupyterIcon;
-      }
-      return (
-        spec.resources['logo-svg'] ||
-        spec.resources['logo-64x64'] ||
-        spec.resources['logo-32x32']
-      );
+      return kernelIcon;
     }
 
-    label(): string {
-      const { kernel, spec } = this;
-      return spec?.display_name || kernel.name;
+    label(): ReactNode {
+      const { kernel } = this;
+      const children = this.children;
+      const summary =
+        children.length == 1
+          ? children[0].label()
+          : this.trans.__(
+              '%1 and %2 more',
+              children[0].label(),
+              children.length - 1
+            );
+      const kernelIdPrefix = kernel.id.split('-')[0];
+      return (
+        <>
+          {summary} <span className={KERNEL_LABEL_ID}>({kernelIdPrefix})</span>
+        </>
+      );
```

Comment:
This breaks `labelTitle` (see line 318 ```const title = [`${this.label()}: ${id}`];```):

![image](https://github.com/jupyterlab/jupyterlab/assets/8435071/4810564a-e0ad-4ce4-9ce6-2a5df80767d7)

Reply:
Thanks, fixed in 5f10b30.